src/main/java/org/ow2/petals/deploymentmodel/ModelDeployer.java

```diff
@@ -0,0 +1,175 @@
+/**
+ * Copyright (c) 2018 Linagora
+ *
+ * This program/library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 2.1 of the License, or (at your
+ * option) any later version.
+ *
+ * This program/library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program/library; If not, see http://www.gnu.org/licenses/
+ * for the GNU Lesser General Public License version 2.1.
+ */
+
+package org.ow2.petals.deploymentmodel;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.nio.file.Files;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+
+import org.apache.commons.io.FileUtils;
+import org.ow2.petals.admin.api.PetalsAdministration;
+import org.ow2.petals.admin.api.PetalsAdministrationFactory;
+import org.ow2.petals.admin.api.artifact.Component.ComponentType;
+import org.ow2.petals.admin.api.artifact.lifecycle.ArtifactLifecycleFactory;
+import org.ow2.petals.admin.api.artifact.lifecycle.ComponentLifecycle;
+import org.ow2.petals.admin.api.artifact.lifecycle.ServiceAssemblyLifecycle;
+import org.ow2.petals.admin.api.exception.ArtifactAdministrationException;
+import org.ow2.petals.admin.api.exception.ArtifactDeployedException;
+import org.ow2.petals.admin.api.exception.ConnectionFailedException;
+import org.ow2.petals.admin.api.exception.ContainerAdministrationException;
+import org.ow2.petals.admin.api.exception.DuplicatedServiceException;
+import org.ow2.petals.admin.api.exception.MissingServiceException;
+import org.ow2.petals.deploymentmodel.busmodel.Bus;
+import org.ow2.petals.deploymentmodel.busmodel.ComponentInstance;
+import org.ow2.petals.deploymentmodel.busmodel.ContainerInstance;
+import org.ow2.petals.deploymentmodel.busmodel.ProvisionedMachine;
+import org.ow2.petals.deploymentmodel.busmodel.ServiceUnitInstance;
+import org.ow2.petals.deploymentmodel.serviceunitmodel.ServiceUnit;
+import org.ow2.petals.jbi.descriptor.JBIDescriptorException;
+import org.ow2.petals.jbi.descriptor.original.JBIDescriptorBuilder;
+import org.ow2.petals.jbi.descriptor.original.generated.Jbi;
+
+public class ModelDeployer {
+
+    private static final Logger LOG = Logger.getLogger(ModelDeployer.class.getName());
```

Comment:
IMO, I think it will be better to build a logger from a logger provided by the client of this library or reuse it directly. The client could use its own logger name for all traces of the library.